### PR TITLE
Fix multi-edge synapses with different synapse types affecting same cell #minor

### PR DESCRIPTION
**Pull request by anilbey**
In multi population circuits, a cell can be affected by multiple edge files. We previously assumed that all synapses affecting a cell had the same properties. However, this isn't always true; for example, plasticity synapses (GLUSYNAPSE) have more properties than GABAAB or AMPANMDA. Mixing GLUSYNAPSE and (GABAAB or AMPANMDA) that are coming from different edge files and affecting the same cell was not supported before. This PR addresses and corrects this assumption.

Reported in #84 

Todos:

- [X] Add type annotations to SynapseFactory
- [x] Add unit tests to SynapseFactory
- [X] Tag the version (via the commit message)

### CODE REVIEW

**Comment by AurelienJaquier** [comment]
small trick: I think you can change the tag of this PR by putting #major or #minor in your commit message: https://github.com/anothrNick/github-tag-action#workflow

File: bluecellulab/circuit/synapse_properties.py [hunk 1]
```diff
@@ -57,6 +58,31 @@ def to_snap(self) -> str:
         return synproperty_to_snap[self]
 
 
+class SynapseProperties:
+    """Synapse Properties to be retrived from circuit and used by the
```
[the remark this answers is not in the document]
Comment (anilbey) [reply bluecellulab/circuit/synapse_properties.py:62]: Thanks

**Comment by anilbey** [comment]
> Nice! Will this have any influence on the e-model-packages synapse plasticity code?

No, this change will only have effect on circuits that are using plasticity-enabled + plasticity-not-enabled synapses together via multiple edge files. e-model-packages circuit with plasticity do not combine multiple edge files having different shapes.

**Comment by anilbey** [comment]
> small trick: I think you can change the tag of this PR by putting #major or #minor in your commit message: https://github.com/anothrNick/github-tag-action#workflow

Wow, this will make my life easier :) Thanks

**Review by AurelienJaquier — APPROVED** [review]
looks good to me!

**Pull request merged**